ibis/pyspark/compiler.py

```diff
@@ -1359,6 +1367,29 @@ def compile_interval_from_integer(t, expr, scope, **kwargs):
 # -------------------------- Array Operations ----------------------------
 
 
+@compiles(ops.ArrayLength)
+def compile_array_length(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    return F.size(src_column)
+
+
+@compiles(ops.ArraySlice)
+def compile_array_slice(t, expr, scope, **kwargs):
```

Comment:
https://spark.apache.org/docs/latest/api/python/pyspark.sql.html#pyspark.sql.functions.slice ?

Reply:
Pyspark `slice()` requires us to specify the length. In Ibis, `end` might be None and length would need to be calculated for every array.

Reply:
anonying, ok this is fine (for now)